lib/MetaProcessor/MetaParser.cpp

```diff
@@ -193,50 +194,66 @@ namespace cling {
         stream = MetaProcessor::kSTDERR;
       // Wrong constant_FD, do not redirect.
       } else if (constant_FD != 1) {
-        llvm::errs() << "cling::MetaParser::isRedirectCommand():"
-                     << "invalid file descriptor number " << constant_FD <<"\n";
+        cling::errs() << "cling::MetaParser::isRedirectCommand():"
+                      << "invalid file descriptor number " << constant_FD <<"\n";
         return true;
       }
       consumeToken();
     }
     // &> redirection for both stdout & stderr
     if (getCurTok().is(tok::ampersand)) {
-      if (constant_FD != 2) {
+      if (constant_FD == 0) {
         stream = MetaProcessor::kSTDBOTH;
       }
       consumeToken();
     }
+    llvm::StringRef file;
     if (getCurTok().is(tok::greater)) {
       bool append = false;
-      consumeToken();
+      // check whether we have >>
+      if (lookAhead(1).is(tok::greater)) {
+        consumeToken();
+        append = true;
+      }
       // check for syntax like: 2>&1
-      if (getCurTok().is(tok::ampersand)) {
-        if (constant_FD != 2) {
+      if (lookAhead(1).is(tok::ampersand)) {
+        if (constant_FD == 0)
           stream = MetaProcessor::kSTDBOTH;
-        }
-        consumeToken();
-      } else {
-        // check whether we have >>
-        if (getCurTok().is(tok::greater)) {
-          append = true;
-          consumeToken();
+
+        const Token& Tok = lookAhead(2);
+        if (Tok.is(tok::constant)) {
+          switch (Tok.getConstant()) {
+            case 1: file = llvm::StringRef("&1"); break;
```

Comment:
It's a const char*, those are stored in the text section. The pointer to it has the scope.